api/app/GraphQL/Mutations/ExtendPoolAdvertisementClosingDate.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\Pool;
+use GraphQL\Error\FormattedError;
+use Illuminate\Support\Carbon;
+
+final class ExtendPoolAdvertisementClosingDate
+{
+    /**
+     * Extends the pool advertisements closing date.
+     * @param  null  $_
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        $poolAdvertisement = Pool::find($args['id']);
+        $oldExpiryDate = $poolAdvertisement->expiry_date;
+        $newExpiryDate = $args['new_expiry_date'];
+
+        if ($newExpiryDate < Carbon::now()->endOfDay() ||
+            $newExpiryDate <= $oldExpiryDate) {
+            throw FormattedError::setInternalErrorMessage("New expiry date must be in the future and greater than the old expiry date.");
```

Comment:
I wonder if this should be a more general "change expiry date" mutation that would allow the user to set any future date.  Requiring the new expiry date to be greater than the old expiry date seems not very useful for the API layer.

Reply:
Ya I think you're right. I guess I was hooked on the word "extend", but having just a change date mutation makes more sense to me.